pages/glossary/highest-possible-contrast.md

```diff
@@ -0,0 +1,6 @@
+---
+title: Highest Possible Contrast
+key: highest-possible-contrast
+---
+
+The highest value when computing the [contrast ratio](https://www.w3.org/TR/WCAG21/#dfn-contrast-ratio) between two sets of colors.
```

Comment:
"the highest value between the two colors", might read as "a value between two other values"

When you want to say the highest value of two values... I think that is what this is trying to say.

Reply:
No, it's not the highest value between two values. It's the highest value you get from computing all the contrast ratios between all the foreground colors and all the background colors. I believe this is correct.

Reply:
> It's the highest value you get from computing all the contrast ratios between all the foreground colors and all the background colors.

I agree that the definition means this.

---
A consequence is that something that is partly black on black and partly white on white has a set of foreground colours of {black, white}, and a set of background colours of {black, white}, thus the "highest possible contrast" is 21:1.

Which can get hairy… Consider an example like this:
```html
    <style>
          .gradient {
            font-size: 72px;
            background: -webkit-linear-gradient(white, black);
            -webkit-background-clip: text;
            -webkit-text-fill-color: transparent;
          }
          .background {
            background: -webkit-linear-gradient(white, black);
          }
        </style>
    <div class="background">
    <div class="gradient">
      Foo
    </div>
  </div>
```
The set of foreground colours ranges from `#000` to `#FFF`. The set of background colours ranges from `#000` to `#FFF` (a bit less because the bounding boxes of the characters are smaller but that's detail) (and the text is visible precisely because the bounding boxes are a bit smaller). So, the "highest possible contrast" between these two sets is pretty close to 21:1 and the rule passes but the SC completely fails.

---
Well, on a purely technical note, the text node is not visible, only its containing `div` is visible (https://github.com/act-rules/act-rules.github.io/issues/977#issuecomment-552831340) but I think this is an issue with the definition of visible… (and this text node should be considered visible).

---
Granted, this is a `webkit` property and thus technology specific, but given that it's working with Chrome, Firefox and Edge, and should be working (not tested) with Safari and Opera, that's pretty annoying.